lib/modules/manager/composer/utils.ts

```diff
@@ -75,7 +75,22 @@ export function extractConstraints(
 
   // extract php
   if (composerJson.config?.platform?.php) {
-    res.php = composerJson.config.platform.php;
+    const major = api.getMajor(composerJson.config.platform.php);
+    const minor = api.getMinor(composerJson.config.platform.php);
+    const patch = api.getPatch(composerJson.config.platform.php);
+
+    // handle where x is taken by composer to mean lowest patch of the lowest minor (x.0.0)
+    if (minor === null || minor === 0) {
+      res.php = `${major}.0.0`;
+    }
+    // handle where x.x is taken by composer to mean lowest patch (x.x.0)
+    else if (patch === null || patch === 0) {
```

Comment:
This covers both 7.2 and 7.2.0. Intended?

Reply:
Correct, if the patch number is missing or already 0, then it will be set to 0. This covers the  first difference where:
NPM: `7.4` => `7.4.*`
Composer: `7.4` => `7.4.0`

Reply:
I decided to use a much simpler approach and added tests to confirm these use cases are covered